src/runtime/getPositions.ts

```diff
@@ -454,27 +454,40 @@ export async function getPositions(
       position.tokens.map((token) => addSourceAppId(token, position.appId)),
     )
 
-    logger.debug({ allTokenDefinitions }, 'allTokenDefinitions')
+    logger.debug(
+      { allTokenDefinitions, count: allTokenDefinitions.length },
+      'allTokenDefinitions',
+    )
 
     // Get the tokens definitions for which we don't have the base token info or position definition
-    const unresolvedTokenDefinitions = allTokenDefinitions.filter(
-      (definition) => {
-        const definitionTokenId = getTokenId({
-          networkId: definition.networkId,
-          address: definition.address,
-        })
-        return (
-          !{ ...baseTokensInfo, ...unlistedBaseTokensInfo }[
-            definitionTokenId
-          ] &&
-          (!visitedDefinitions[definitionTokenId] ||
-            visitedDefinitions[definitionTokenId]?.type !==
-              'app-token-definition')
-        )
-      },
+    const unresolvedTokenDefinitionsByTokenId: Record<
+      string,
+      (typeof allTokenDefinitions)[number]
+    > = {}
+    for (const tokenDefinition of allTokenDefinitions) {
+      const definitionTokenId = getTokenId({
+        networkId: tokenDefinition.networkId,
+        address: tokenDefinition.address,
+      })
+      if (
+        baseTokensInfo[definitionTokenId] ||
+        unlistedBaseTokensInfo[definitionTokenId] ||
+        visitedDefinitions[definitionTokenId]?.type ===
+          'app-token-definition' ||
+        unresolvedTokenDefinitionsByTokenId[definitionTokenId]
+      ) {
+        continue
+      }
+      unresolvedTokenDefinitionsByTokenId[definitionTokenId] = tokenDefinition
+    }
```

Comment:
Updated the logic so we deduplicate the token definition.
which ends up saving on RPC calls too.